pkg/imexport/export.go

```diff
@@ -50,8 +54,95 @@ func createDir(name string, tw *tar.Writer, dir *vfs.DirDoc) error {
 	return err
 }
 
-func export(tw *tar.Writer, fs vfs.VFS) error {
-	root := "/Documents"
+func metadata(tw *tar.Writer, fs vfs.VFS, domain string) error {
+	db := couchdb.SimpleDatabasePrefix(domain)
+	doctype := "io.cozy.files"
+	req := &couchdb.AllDocsRequest{}
+	var results []map[string]interface{}
+	if err := couchdb.GetAllDocs(db, doctype, req, &results); err != nil {
+		return err
+	}
+
+	metaDir := "metadata/album/"
+	hdrDir := &tar.Header{
+		Name:     metaDir,
+		Mode:     0755,
+		Typeflag: tar.TypeDir,
+	}
+	if err := tw.WriteHeader(hdrDir); err != nil {
+		return err
+	}
+
+	hdr := &tar.Header{
+		Name:       metaDir + "album.json",
+		Mode:       0644,
+		AccessTime: time.Now(),
+		ChangeTime: time.Now(),
+		Typeflag:   tar.TypeReg,
+	}
+
+	var content bytes.Buffer
+	var size int64
+
+	for _, val := range results {
+		if val["referenced_by"] != nil {
+
+			id := val["_id"].(string)
+			doc, err := fs.FileByID(id)
+			if err != nil {
+				return err
+			}
+
+			path, err := fs.FilePath(doc)
+			if err != nil {
+				return err
+			}
+			val["album_filepath"] = path
+
+			ref := doc.ReferencedBy
+			for _, v := range ref {
+				out := &couchdb.JSONDoc{}
+				if err = couchdb.GetDoc(db, v.Type, v.ID, out); err != nil {
+					return err
+				}
+				m := out.ToMapWithType()
+
+				albumName := m["name"]
+				val["album_name"] = albumName
+			}
+
+			jsonString, err := json.Marshal(val)
```

Comment:
`val` contains the entire file object, which we already have in the files folder of the tarball. It would be more efficient to only save here the fields from the `io.cozy.album`,  `m` in your code. 

Careful, do not assume all `referenced_by` will be to albums, for now you can discard other types, but the export should not break when we introduce other references.